ocamltest/run_win32.c

```diff
@@ -26,33 +26,38 @@
 #include <stdarg.h>
 #include <sys/types.h>
 
+#include "caml/osdeps.h"
+
 #include "run.h"
 #include "run_common.h"
 
 static void report_error(
   const char *file, int line,
   const command_settings *settings,
-  const char *message, const char *argument)
+  const char *message, const WCHAR *argument)
 {
-  char error_message[1024];
+  WCHAR error_message[1024];
   DWORD error = GetLastError();
+  char *error_message_c;
   FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM, NULL, error, 0,
-    (LPTSTR) &error_message, sizeof(error_message), NULL);
+                error_message, sizeof(error_message)/sizeof(WCHAR), NULL);
+  error_message_c = caml_stat_strdup_of_utf16(error_message);
   if ( is_defined(argument) )
     error_with_location(file, line,
-      settings, "%s %s: %s", message, argument, error_message);
+      settings, "%s %s: %s", message, argument, error_message_c);
   else
     error_with_location(file, line,
-      settings, "%s: %s", message, error_message);
+      settings, "%s: %s", message, error_message_c);
+  caml_stat_free(error_message);
```

Comment:
AAAARGH!  You're freeing the wrong error_message!!

Reply:
Thanks, how embarrassing! (there should be a warning somewhere about writing C with less than 5 hours of sleep...)